include/reactor-uc/macros.h

```diff
@@ -188,18 +210,34 @@
     Reaction *effects[(EffectSize)];                                                                                   \
   } ActionName;
 
-#define DEFINE_ACTION_CTOR_FIXED(ActionName, ActionType, EffectSize, SourceSize, BufferType, BufferSize, MinDelay)     \
+#define GET_ARG9(arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9, ...) arg9
+#define GET_ARG8(arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, ...) arg8
+#define GET_ARG7(arg1, arg2, arg3, arg4, arg5, arg6, arg7, ...) arg7
+#define GET_ARG6(arg1, arg2, arg3, arg4, arg5, arg6, ...) arg6
+
+#define DEFINE_ACTION_STRUCT_CHOOSER(...)                                                                              \
+  GET_ARG7(__VA_ARGS__, DEFINE_ACTION_STRUCT_WITH_VALUE, NULL, DEFINE_ACTION_STRUCT_WITHOUT_VALUE)
```

Comment:
Would be great with a short descripbtion of why it works (and its assumptions on the parmeters to DEFINE_ACTION_STRUCT.